pkg/controller/pravega/pravega_segmentstore_test.go

```diff
@@ -356,10 +371,34 @@ var _ = Describe("PravegaSegmentstore", func() {
 					_ = pravega.MakeSegmentStoreExternalServices(p)
 					Ω(err).Should(BeNil())
 				})
+			})
+			Context("Create External service with external service type and access type empty", func() {
+				BeforeEach(func() {
+					p.Spec.Pravega.SegmentStoreExternalServiceType = ""
+					p.Spec.ExternalAccess.Type = ""
+					p.Spec.ExternalAccess.DomainName = "example"
+				})
+				It("should create external service", func() {
 
+					_ = pravega.MakeSegmentStoreExternalServices(p)
+					Ω(err).Should(BeNil())
+				})
 			})
+			Context("Create External service with external service type empty and access type load balancer", func() {
+				BeforeEach(func() {
+					m := make(map[string]string)
+					p.Spec.Pravega.SegmentStoreServiceAnnotations = m
+					p.Spec.Pravega.SegmentStoreExternalServiceType = ""
+					p.Spec.ExternalAccess.Type = corev1.ServiceTypeLoadBalancer
```

Comment:
SegmentStoreService type was anyways set to LoadBalancer in the BeforeEach. So lets not set ExternalAccess type to LoadBalancer here. Lets check if it is taking the Service type to the value of ExternalAccess.Type which is set to ClusterIP in the BeforeEach

Reply:
intent is to just to cover service is getting created in these case.

Reply:
Got it now, i have added the checks